compiler/qsc_qasm3/src/angle.rs

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#[cfg(test)]
+pub(crate) mod tests;
+
+use num_bigint::BigInt;
+
+use crate::oqasm_helpers::safe_u64_to_f64;
+use std::convert::TryInto;
+use std::f64::consts::PI;
+use std::fmt;
+use std::ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Sub, SubAssign};
+
+/// A fixed-point angle type with a specified number of bits.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+struct Angle {
+    value: u64,
+    size: u32,
+}
+
+#[allow(dead_code)]
```

Comment:
what's the need for allowing dead code here?

Reply:
Note that you can add a reason to the allow attribute,
```rust
#[allow(dead_code, reason = "this will support X future qiskit feature")]
```

Reply:
Lint reasons are experimental